### DISCUSSION TITLE
Outdoor Scene Based on LIDAR data

### DISCUSSION THREAD

**Discussion by manoj-kumar-joshi** [NVlabs/sionna#265] [Show and tell]
I am working on a project to create a 3D scene compatible with Sionna using LIDAR data. Attaching images of output (Left image is from Google 3D tiles and right one is generated using "Render" function. any comments or suggestions are welcome. I can share the scene for others to play around if needed.

![image](https://github.com/NVlabs/sionna/assets/61615138/25177cae-567d-4fff-8640-48ae016d8aff)
![image](https://github.com/NVlabs/sionna/assets/61615138/3fefa62a-df5e-4e57-b3a8-e79344dd7680)

**Comment by faycalaa** [comment]
Hi,

This looks like an interesting project.

Currently, the ray tracer might struggle with specular reflections when meshes are too detailed, i.e., made of too many triangles. It may be beneficial to simplify the object geometries. For example, rectangles used for modeling the walls should ideally be comprised of just two triangles each.

Also, keep in mind that Sionna does not currently model foliage. This means that interactions with tree elements might not be accurately represented.

**Comment by manoj-kumar-joshi** [comment]
Thank you for your comments. Bellow is the wireframe of one of the buildings (To give you an idea of triangles)
![image](https://github.com/NVlabs/sionna/assets/61615138/8dd45c21-6024-452e-b312-5a029d950c34)

I have also generated coverage map for the scene.
![image](https://github.com/NVlabs/sionna/assets/61615138/72dea3ec-2b5b-457c-9a21-2fbfa8160bfd)

For Foliage, I was planning to use a custom material with very less reflection and large scattering (Not figured out yet how to do though).

**Reply by jhoydis** [reply]
One issue with this approach is that foliage cannot be penetrated, i.e., rays cannot go through trees. This might lead to overly pessimistic pathloss predictions.

It is great to see that coverage map computation seems to work fine. However, computing channel impulse responses might be challenging as @faycalaa said above because it becomes computationally demanding to find exact specular paths when a scene consists of two many triangles. Could you try to compute channel impulse responses to a few receivers and show the found paths (using `scene.render(paths=paths)`)?

**Reply by manoj-kumar-joshi** [reply]
OK. Attaching a path and CIR for the same receiver (I kept max depth as 10). Please also note that I have not applied lookat position of transmitter to receiver.
![image](https://github.com/NVlabs/sionna/assets/61615138/00ab2b44-459c-46dd-9870-f2e9ef8ab2d2)
![image](https://github.com/NVlabs/sionna/assets/61615138/759aee26-ee94-47de-9108-3935c1bf0845)

**Reply by manoj-kumar-joshi** [reply]
I also tried increasing the num_samples=5000000 but paths remain same. I can try to reduce number of triangles in building walls.

**Reply by fanyimeng0** [reply]
Hello, I would like to know how you export your LiDAR data and how you assign material properties to it. Have you considered using the LiDAR's RCS (Radar Cross Section) values to assist in setting the materials? Thank you.

**Reply by manoj-kumar-joshi** [reply]
Hello , I have created a LIDAR processing pipeline where I do segmentation, mesh creation for trees, ground and buildings. After that I assign martials in another python script.

**Reply by ifelse1984** [reply]
This is very interesting,  thank you for sharing! Would you be able to share the software/code/resources for how to do this? And just wondering if this works if the lidar inputs might lead to non-watertight shapes (e.g. lidar scans from an autonomous vehicle, so the details behind objects are not known)?

**Reply by sathibabu9** [reply]
Good morning sir,thank you for sharing valuable results about creating our own scene ,if possible can you ples guide me how to do this 

best regards 
satyam
SRM University
